etp-core/etp-backend/src/test/clj/solita/etp/valvonta_kaytto/valvonta_csv_test.clj

```diff
@@ -0,0 +1,411 @@
+      (jdbc/insert! ts/*db* :vk_toimenpide {:valvonta_id  valvonta-id
+                                            :type_id      0
+                                            :create_time  start-timestamp
+                                            :publish_time start-timestamp
+                                            :diaarinumero "ARA-05.03.01-2024-159"})
+
+      ;; Add kehotus-toimenpide to the valvonta
+      (jdbc/insert! ts/*db* :vk_toimenpide {:valvonta_id   valvonta-id
+                                            :type_id       2
+                                            :create_time   kehotus-timestamp
+                                            :publish_time  kehotus-timestamp
+                                            :deadline_date kehotus-deadline
+                                            :diaarinumero  "ARA-05.03.01-2024-159"})
+
+      ;; Create energiatodistus and sign it, the signing time is on the same day as the kehotus deadline
+      (let [todistus (-> (test-data.energiatodistus/generate-add 2018 true)
+                         (assoc-in [:perustiedot :rakennustunnus] rakennustunnus))
+            todistus-id (test-data.energiatodistus/insert! [todistus] laatija-id)]
+        (test-data.energiatodistus/sign-at-time! todistus-id
+                                                 laatija-id
+                                                 true
+                                                 (-> kehotus-deadline
+                                                     (.atStartOfDay (ZoneId/systemDefault))
+                                                     .toInstant)))
+
+      ;; Close the valvonta
+      (jdbc/insert! ts/*db* :vk_toimenpide {:valvonta_id  valvonta-id
+                                            :type_id      5
+                                            :create_time  close-timestamp
+                                            :publish_time close-timestamp
+                                            :diaarinumero "ARA-05.03.01-2024-159"})
+
+      (create-csv (partial handle-output call-count output))
+
+      (t/is (= 4 @call-count))
+      (t/is (= @output
+               [csv-header-line
+                "1;\"3139000812\";\"ARA-05.03.01-2024-159\";\"Testitie 5\";\"90100\";\"OULU\";1;\"Valvonnan aloitus\";2024-03-18T02:00;\"Tuntija, Asian\";\n"
+                "1;\"3139000812\";\"ARA-05.03.01-2024-159\";\"Testitie 5\";\"90100\";\"OULU\";2;\"Kehotus\";2024-03-20T02:00;\"Tuntija, Asian\";\"x\"\n"
+                "1;\"3139000812\";\"ARA-05.03.01-2024-159\";\"Testitie 5\";\"90100\";\"OULU\";3;\"Valvonnan lopetus\";2024-03-30T02:00;\"Tuntija, Asian\";\n"])
+            "All toimenpiteet for the valvonta are present, energiatodistus is marked being created during kehotus toimenpide"))))
+
+(t/deftest energiatodistus-hankittu-on-the-day-after-deadline-test
+  (t/testing "Energiatodistus hankkittu is not set for kehotus when the todistus was acquired the day after the deadline"
```

Comment:
Onko tämä haluttu toimivan näin? Energiatodistus ei näy CSV:ssä ollenkaan?

Reply:
Nähtävästi on päätellen seuraavasta testistä, mutta hassun kuuloinen

Reply:
Tuskailin tämän kanssa että miten muotoilisin näistä testeistä selkeät. Se tämän alla oleva kommentti pyrkii selittämään tämän osan pointin.

Reply:
Käytännössähän siis tämän deadlinen jälkeen tulleen energiatodistuksen yhteyttä tuohon toimenpiteeseen ei voida tietää kuin vasta sitten, kun seuraava toimenpide on luotu.

Reply:
Milloin se voisi mennä seuraavalle toimenpiteelle?

Reply:
Ei se voikaan mennä, mutta tuota aikavälin loppua ei kuitenkaan voi (tai kannata) laittaa myöskään hamaan tulevaisuuteen. Joka tapauksessa prosessi etenee tuossa aina seuraavaan toimenpiteeseen (todennäköisesti valvonnan lopettamiseen) jonka myötä energiatodistus merkataan hankituksi tuon aiemman toimenpiteen aikana. Eli tuo on väliaikainen tila ennen kuin valvoja jatkaa käsittelyä (==  huomaa energiatodistuksen saapuneen).

Näin minä tämän itselleni järkeilin. En tiedä osasinko kuvata tilanteen testissä tai kirjoittaa auki kovinkaan hyvin tähän.